ADR/adr-0008-secret-mgmt-for-user-workloads.md

```diff
@@ -0,0 +1,60 @@
+# Secret Management For User Workloads
+
+Date: 2022-10-28
+
+## Status
+
+Accepted
+
+## Context
+
+When user workloads are deployed to environments, the system should be able to provide a way to inject values that is specific to the environment. Today, this is done through environment variables that are managed as overlays on the GitOps repository for the application. However, this method does not provide a good way to manage secrets. This ADR addresses the secret management of user workloads for different environments.
+
+## Decision
+
+### Architecture Overview
+
+All user workloads are synced through GitOps service to their environments. StoneSoup's GitOps service uses the [Argo CD Vault Plugin (AVP)](https://github.com/argoproj-labs/argocd-vault-plugin) to replace secret values from the supported secret management backends.
+
+AppStudio will provide a default secret management backend (the default backend) that will be used to store the secrets by default. When a GitOps repository is generated by the application service templates should already be ready for secret use.
+![](../diagrams/secret-mgmt.excalidraw.svg)
+
+#### Secret management APIs
+An API is provided to create and update secrets on the default backend. This is a REST-based API similar to [SPI token management API](https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#post-tokennamespacename). The secrets are per component and can be varied per environment. Only workspace owners can create and update secrets.
+
+#### The default secret management backend
+**TODO:** Add the default backend and its deployment shape.
+
+#### Bring Your Own Secret Management
+Applications can use secret management backends other than the default secret management backend. However, this requires the GitOps service to provide single-tenant ArgoCD instances. The supported backends will be limited to those supported by [AVP as documented](https://argocd-vault-plugin.readthedocs.io/en/stable/backends/). Creating and updating secrets through StoneSoup secret management REST APIs is not supported for custom backends.
```

Comment:
A concern that I have with using a plugin such as AVP in combination with BYO-Secret-Manager is, that the users leveraging their own secrets manager would a) have to give us credentials for read access to their secrets management system and b) entrust us with the knowledge about the secrets during the processing of manifests.

Reply:
Bringing your own is not an immediate requirement, it is more of a thought exercise. We can remove it for now.

Reply:
I think we are moving ahead *without* the plugin - exploring new designs now. Closing this.